llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -560,7 +560,7 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
       })
       .widenScalarOrEltToNextPow2(1)
       .clampScalar(0, s32, s32)
-      .clampScalarOrElt(1, MinFPScalar, s64)
+      .clampScalarOrElt(1, MinFPScalar, s128)
```

Comment:
Is this needed? If so could it use minScalarOrElt?

Is it worth scalarizing vectors in this same patch?

Reply:
> Is this needed? If so could it use minScalarOrElt?

Replaced with `minScalarOrElt`.

> Is it worth scalarizing vectors in this same patch?

Yes, added `.scalarizeIf(scalarOrEltWiderThan(1, 64), 1)`. 

`v3f128_fp128` is still failing because we failed to legalize `%21:_(<3 x s128>) = G_OR %29:_, %30:_`